docs/developer-docs/security/formal-verification.md

```diff
@@ -0,0 +1,26 @@
+---
+sidebar_position: 4
+sidebar_label: Formal verification
+---
+# Formal verification
+
+Formal verification is, arguably, the highest form of quality assurance for software. Given a specification of what the system should do, formal verification tools check whether this specification is satisfied by a model of the system. The unique advantage of formal verification is that it can not only find bugs but also formally **prove** their absence – including the absence of security bugs. This goes beyond what testing or manual audits can achieve.
+
+The proof is however, always relative to the model and the specification. Any simplifications and assumptions in the model, or omissions in the specification, may hide bugs and attacks. However, verification can require a lot of effort, and model simplifications can make it significantly easier. For a concrete example, when verifying the ckBTC minter canisters, the DFINITY team used  models that exclude the possibility of concurrently calling the `update_balance` canister method more than 2 times. This potentially misses attacks that require 3 or more concurrent calls to `update_balance` to trigger a bug. But we considered such bugs highly unlikely and, in return, we were able to run a fully automatic verification process, which was much cheaper than other verification methods.
```

Comment:
```suggestion
The proof is relative to the model and the specification. Any simplifications and assumptions in the model, or omissions in the specification, may hide bugs and attacks. Verification can require a lot of effort, and model simplifications can make it significantly easier. 

For a concrete example, when verifying the ckBTC minter canisters, the DFINITY team used models that exclude the possibility of concurrently calling the `update_balance` canister method more than 2 times. This potentially misses attacks that require 3 or more concurrent calls to `update_balance` to trigger a bug. The team considered such bugs highly unlikely and, in return, we were able to run a fully automatic verification process, which was much cheaper than other verification methods.
```

Reply:
I'd also like to "however" or "on the other hand", as it's showing that there's a tradeoff (better guarantees vs ease of application).